### PR TITLE
[ALLUXIO-2892] Support alias for Alluxio configuration

### PR DESCRIPTION
**Pull request by cmuyaol**
https://alluxio.atlassian.net/browse/ALLUXIO-2892

### CODE REVIEW

**Comment by apc999** [comment]
unnecessary file mode change for
`core/common/src/main/java/alluxio/PropertyKey.java` from 100644 to 100755

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 1]
```diff
@@ -1302,6 +1302,24 @@ static PropertyKey create(Template template, Object defaultValue, Object... para
...
+   * Factory method to create a constant default property
+   * and assign a default value together with its alias.
+   *
+   * @param name String of this property
+   * @param defaultValue Default value of this property in compile time if not null
+   * @param aliases String list of aliases of this property
+   */
+  static PropertyKey create(String name, Object defaultValue, String[] aliases) {
+    PropertyKey key = new PropertyKey(name);
+    for (String alias : aliases) {
```
Comment (apc999) [inline core/common/src/main/java/alluxio/PropertyKey.java]: Better check if `alias` is null or not.

File: core/common/src/test/java/alluxio/PropertyKeyTest.java [hunk 2]
```diff
@@ -58,6 +67,11 @@ public void isValid() throws Exception {
   }
 
   @Test
+  public void aliasIsValid() throws Exception {
+    Assert.assertTrue(PropertyKey.isValid(sDebugAlias.toString()));
```
Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: NIT: `Assert.assertTrue` -> `assertTrue` as `Assert.assertTrue` has been statically imported (line 15)

File: core/common/src/test/java/alluxio/PropertyKeyTest.java [hunk 3]
```diff
@@ -18,18 +18,27 @@
...
 
 import org.junit.Assert;
+import org.junit.BeforeClass;
 import org.junit.Test;
 
 /**
  * Tests enum type {@link PropertyKey}.
  */
 public final class PropertyKeyTest {
+  static PropertyKey sDebugAlias;
```
Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: - let's not mess up with `PropertyKey.DEBUG`, instead, we can simply create a test property.
- why static member variable? can we simply have the following definition and initialization?
```java
private PropertyKey mTestProperty;

@Before
public void before() {
  mTestProperty = PropertyKey.create("alluxio.test.property", false, new String[] {"alluxio.test.alias1", "alluxio.test.alias2"});
}
```

File: core/common/src/test/java/alluxio/PropertyKeyTest.java [hunk 4]
```diff
@@ -18,18 +18,27 @@
...
+  public static void initialize() {
+    sDebugAlias = PropertyKey.create(PropertyKey.Name.DEBUG, false,
+            new String[] {"alluxio.debug.alias1", "alluxio.debug.alias2"});
+  }
   /**
    * Tests parsing string to PropertyKey by {@link PropertyKey#fromString}.
    */
   @Test
   public void fromString() throws Exception {
+    assertEquals(PropertyKey.DEBUG, PropertyKey.fromString("alluxio.debug.alias1"));
```
Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: also test the other alias?

File: core/common/src/test/java/alluxio/PropertyKeyTest.java (hunk 2 above)
Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: test the other alias is also valid?

File: core/common/src/main/java/alluxio/PropertyKey.java [hunk 5]
```diff
@@ -1302,6 +1302,24 @@ static PropertyKey create(Template template, Object defaultValue, Object... para
...
+  /**
+   * Factory method to create a constant default property
+   * and assign a default value together with its alias.
+   *
+   * @param name String of this property
+   * @param defaultValue Default value of this property in compile time if not null
+   * @param aliases String list of aliases of this property
+   */
+  static PropertyKey create(String name, Object defaultValue, String[] aliases) {
+    PropertyKey key = new PropertyKey(name);
```
Comment (apc999) [inline core/common/src/main/java/alluxio/PropertyKey.java]: To make this simple, 
```java
PropertyKey key = create(name, defaultValue);
```
It is a good practice to re-use existing factory methods/constructors as much as possible and leave only the logic that is really different here. It is easier to understand and maintain too.

File: core/common/src/test/java/alluxio/PropertyKeyTest.java [hunk 6]
```diff
@@ -18,19 +18,33 @@
...
 
 /**
  * Tests enum type {@link PropertyKey}.
  */
 public final class PropertyKeyTest {
+  private PropertyKey mTestProperty;
+  private PropertyKey mNullAliasTestProperty;
 
+  @Before
+  public void initialize() {
```
Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: nit: this is the coding convention in Alluxio:

```java
@Before
  public void before() {
    ...
  }
```

File: core/common/src/test/java/alluxio/PropertyKeyTest.java [hunk 7]
```diff
@@ -18,19 +18,33 @@
...
+    mTestProperty = PropertyKey.create("alluxio.test.property", false,
```
Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: NIT: for `mTestProperty` and `mNullAliasTestProperty`, you can initialize them directly along with the definition without adding this `Before` method. e.g.:

```java
  private PropertyKey mTestProperty = PropertyKey.create("alluxio.test.property", false,
      new String[] {"alluxio.test.property.alias1", "alluxio.test.property.alias2"});
  private PropertyKey mNullAliasTestProperty = 
      PropertyKey.create("alluxio.test.property.nullAlias", false, null);
```

File: core/common/src/test/java/alluxio/PropertyKeyTest.java [hunk 8]
```diff
@@ -58,6 +72,12 @@ public void isValid() throws Exception {
   }
 
   @Test
+  public void aliasIsValid() throws Exception {
+    assertTrue(PropertyKey.isValid(mTestProperty.toString()));
+    assertTrue(PropertyKey.isValid(mNullAliasTestProperty.toString()));
```
Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: not sure what are we testing here? could you explain this? why this line 77 is related to alias?

Comment (cmuyaol) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: I am sorry, I was intended to test if the isValid method will work for the alias, and check if null will not be as a valid key, but I forgot to change the code. Here is the corresponding code:
`assertTrue(PropertyKey.isValid("alluxio.test.property.alias1"));
  assertTrue(PropertyKey.isValid("alluxio.test.property.alias2"));
  assertFalse(PropertyKey.isValid(null));`
Is that ok?

Comment (cmuyaol) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: And, should we consider the situation if the input of the isValid and fromString method is null?
When I run the local test I found if you give the null as input you will get a nullPointerException.

Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: - this is good:
```java
assertTrue(PropertyKey.isValid("alluxio.test.property.alias1")); assertTrue(PropertyKey.isValid("alluxio.test.property.alias2")); assertFalse(PropertyKey.isValid(null));
```
- Good to have null check in `isValid` and `fromString`. You can submit separate PRs on it as it looks less related to this PR. thanks

File: core/common/src/test/java/alluxio/PropertyKeyTest.java [hunk 9]
```diff
@@ -25,12 +25,19 @@
  */
 public final class PropertyKeyTest {
 
+  private PropertyKey mTestProperty = PropertyKey.create("alluxio.test.property", false,
+            new String[] {"alluxio.test.property.alias1", "alluxio.test.property.alias2"});
```
Comment (apc999) [inline core/common/src/test/java/alluxio/PropertyKeyTest.java]: NIT: indentation is off at line 29. 4-space indent
```java
private PropertyKey mTestProperty = PropertyKey.create("alluxio.test.property", false,
     new String[] {"alluxio.test.property.alias1", "alluxio.test.property.alias2"});
```
